File: my-messenger-android/app/src/main/java/my/messenger/androidclient/api/model/UserProfile.java
package my.messenger.androidclient.api.model;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Date;

public class UserProfile implements Serializable {

    private String id;
    private String username;
    private String password;
    //@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date birthDate;

    public UserProfile() {

    }

    public UserProfile(String id, String username) {
        this.id = id;
        this.username = username;
    }

    public UserProfile(String id, String username, String password, Date birthDate) {
        this(id, username);
        this.password = password;
        this.birthDate = birthDate;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    public Date getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    @Override
    public String toString() {
        return "UserProfile{" +
                "id='" + id + '\'' +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                ", birthDate=" + birthDate +
                '}';
    }
}

File: my-messenger-backend/my.messenger.common/Users/UserProfile.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace my.messenger.common.Users
{
    public class UserProfile
    {
        public String Id { get; set; }
        public String Username { get; set; }
        public String Password { get; set; }
        public DateTime BirthDate { get; set; }

        public UserProfile(string id, string username, string password, DateTime birthDate)
        {
            Id = id;
            Username = username;
            Password = password;
            BirthDate = birthDate;
        }

        public UserProfile()
        {
            Id = Identifier.New();
        }
    }
}
